securesystemslib/hsm_keys.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+
+"""
+<Program Name>
+  hsm_keys.py
+
+<Author>
+  Tanishq Jasoria <jasoriatanishq@gmail.com>
+
+<Purpose>
+  This module provides a high-level API for using hardware security modules
+  for various cryptographic operations
```

Comment:
What's the rationale for having both `hsm_keys` and `hsm` modules? The high-level interface that may be called by a developer, who uses `securesystemslib`, should be in the `keys` module.

Please also make sure that the high-level interface provide data such es created signatures or exported public keys in an expected format (see `securesystemslib.formats.{*KEY_SCHEMA, *SIGNATURE_SCHEMA}`).

Reply:
When I started I was not sure which library to use to interact with the hardware tokens. So, I thought of having two modules, one which implement basic function to interact with the tokens, while hiding the library specific details and the other module will use these function to incorporate required functionality in securesystemslib. 
Now if we decide not to use PyKCS11, we can simply modify the hsm.py module without making major changes in the hsm_keys.py module.

Reply:
Fair enough. Please feel free to chime in on the discussion about `securesystemslib`'s API in #35. IMO the different layers of abstraction should be more intuitive or more clearly defined (or both). But that's not something we need to fix here.